### DISCUSSION TITLE
Open3D window management - manage multiple windows and embed in custom application

### DISCUSSION THREAD

**Discussion by AdvikMehta** [isl-org/Open3D#6297] [Q&A]
Hello everyone, I am currently developing a project with open3d for manually registering and processing point clouds. 

I am following [this example](http://www.open3d.org/docs/release/tutorial/visualization/interactive_visualization.html) from the open3D documentation. Here, we use manual registration to pick points in the 2 clouds manually and register with ICP for refinement. However, the point selection of clouds occurs one after the other. 

In my application, I would like to display two windows (or multiple viewports per window) in order for a better experience while picking points. Moreover, I would like to display the window generated by open3d to be captured and shown in my application which will have a GUI / action bar and other features.

Is there a way to achieve this, ie, display multiple windows for point selection and capture these windows and display them in my own external app (built in Java, subject to change based on this problem haha)?

**Comment by ssheorey** [comment]
Hi @AdvikMehta check out this example for a GUI with multiple windows:

http://www.open3d.org/docs/release/python_example/visualization/index.html#multiple-windows-py

See the GUI elements available in the `o3d.visualization.gui` namespace

Note that you should not mix this new GUI based visualization (open3d.visualization.{gui,rendering}) namespaces with the legacy visualization methods (open3d.visualization.visualizer*). Also, Open3D doesn't have Java bindings - only C++ and Python.
